src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,314 @@
+    now1    = [NSDate date];
+    
+    if (nil == self.locationManager)
+        self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    
+    
+    [self.locationManager startMonitoringSignificantLocationChanges];
+    [self.locationManager startUpdatingLocation];
+    [self regionStart];
+}
+
+
+// recieve GPS updated location
+- (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations {
+    
+    
+    location = [locations lastObject];
+    
+    NSDate* timestamp = location.timestamp;
+    
+    
+    
+    NSTimeInterval age = [NSDate.date timeIntervalSinceDate:location.timestamp];
+    
+    if (fabs([now timeIntervalSinceDate:timestamp]) < 60){
+        
+        NSLog(@"Time 60 second print");
+        
+        if (fabs([now1 timeIntervalSinceDate:timestamp]) > 15) {
+            
+            now1 = timestamp;
+
+            
+            DateFormatter = [[NSDateFormatter alloc] init];
+            [DateFormatter setDateFormat:@"yyyyMMdd'T'HH:mm:ss"];
+            [self InsertData];
+        }
+        
+    }
+    else{
+        NSLog(@"Time 60 more than print");
+        [self MakeJSON_Make ];
+        [self stopUpdatingLocationWithMessage:NSLocalizedString(@"Error", @"Error")];
+        
+    }
+}
+
+
+
+- (void)stopUpdatingLocationWithMessage:(NSString *)state {
+    
+    [self.locationManager stopUpdatingLocation];
+    self.locationManager.delegate = nil;
+}
+
+-(void)regionStart{
+    NSArray *ns_coord = [coord componentsSeparatedByString:@","];
+    
+    NSString *str_lat = [NSString stringWithFormat:@"%@", ns_coord[1]];
+    float lat = [str_lat floatValue];
+    
+    NSString *str_lon = [NSString stringWithFormat:@"%@", ns_coord[2]];
+    float lon = [str_lon floatValue];
+    
+    NSString *str_radio = [NSString stringWithFormat:@"%@", ns_coord[3]];
+    float radio = [str_radio floatValue];
+    
+    
+    CLLocationCoordinate2D center = CLLocationCoordinate2DMake(lat, lon);
+    CLRegion *region = [[CLCircularRegion alloc] initWithCenter:center radius:radio identifier:@"Center"];
+    
+    [self.locationManager startMonitoringForRegion:region];
+}
+
+-(void)locationManager:(CLLocationManager *)manager didStartMonitoringForRegion:(CLRegion *)region{
+    
+    //[self MakeJSON_Make];
+    
+    NSLog(@"Now monitoring for %@", region.identifier);
+    NSLog(@"Welcome to %f", region.radius);
+    NSString *message = @"Show Message...\n didStartMonitoringForRegion";
+    UIAlertView *toast = [[UIAlertView alloc]initWithTitle:nil message:message delegate:nil cancelButtonTitle:nil otherButtonTitles:nil, nil];
+    
+    [toast show];
+    int duration = 1;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, duration *NSEC_PER_SEC), dispatch_get_main_queue(), ^{[toast dismissWithClickedButtonIndex:0 animated:YES];
+    });
+}
+-(void)locationManager:(CLLocationManager *)manager didEnterRegion:(CLRegion *)region{
```

Comment:
didEnterRegion and didExitRegion should  both insert a record into the database (afaics they don't but instead display a UI message?)